### ISSUE TITLE
amplify_codebuild script

### ISSUE THREAD

**Issue by joekiller** [aws-amplify/amplify-console#737]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to test my amplify.yml build prior to pushing it up.

**Describe the solution you'd like**
A script similar to [AWS CodeBuild's codebuild_build](https://github.com/aws/aws-codebuild-docker-images/tree/master/local_builds)

**Additional context**
Thank you for the build image though. That is helpful.

**Comment by hirenumradia** [comment]
+1 for this, it's really frustrating when you have a build error in the amplify build and can't reproduce it locally.

**Comment by hughesjj** [comment]
Any word on this?

**Comment by ahurlburt** [comment]
Agreed, i have a build error I can't reproduce. Seems to be missing a dependency when running npm build. Everything works fine locally.

Are there any suggestions/instructions on how to debug this? If I pull the build image what steps do I run inside the container? I don't even have an amplify.yml in my project dir? I'm sure I could reverse engineer from the logs but if anyone has helpful suggestions it would be appreciated (and I'm sure by anyone else coming across this thread).

**Comment by L14MJ4M3S0N** [comment]
@ahurlburt I had the same issue. It ended up being because of the `NODE_ENV` 
If this is set to `production` before a `yarn install`, yarn will only install deps and not devdeps 
This can be overridden by using `--production=false`

**Comment by jalcock501** [comment]
is there an update on this feature request?
